File: framework/src/main/java/com/software/dm/swallow/stormy/algoac/inter/VagueState.java
package com.software.dm.swallow.stormy.algoac.inter;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;

import com.software.dm.swallow.stormy.algoac.pojo.VagueResultEntity;

/**
 * @param <T>
 * @author DM
 * @version v1.0.0.1
 * @Description
 * @date 2017
 */
public interface VagueState<T> {

    /**
     *
     */
    public static final char STAR_CHAR = '*';

    /**
     * @param t
     * @param nextNormalState
     */
    public void addNextNormalStateMap(T t, VagueState<T> nextNormalState);

    /**
     * @param t
     * @param nextVagueState
     */
    public void addNextVagueStateMap(T t, VagueState<T> nextVagueState);

    /**
     * @return
     */
    public Map<T, VagueState<T>> getNextVagueStateMap();

    /**
     * @return
     */
    public Map<T, VagueState<T>> getNextNormalStateMap();

    /**
     * @param vagueResultEntityList
     */
    public void addVagueResultEntityList(VagueResultEntity vagueResultEntityList);

    /**
     * @return
     */
    public List<VagueResultEntity> getVagueResultEntityList();

    /**
     *
     */
    public Map<String, List<Object>> patternMap = new HashMap<String, List<Object>>();

    /**
     * @param pattern
     * @param vagueResultEntity
     */
    public void addState(final String pattern, final VagueResultEntity vagueResultEntity);

    /**
     * @param patterns
     * @param resultDataSet
     */
    public void serachVagueResult(final String patterns, final Set<Object> resultDataSet);

}

File: framework/src/main/scala/com/software/dm/swallow/stormy/scala/algoac/inter/VagueState.scala
package com.software.dm.swallow.stormy.scala
package algoac
package inter

import pojo.VagueResultEntity

import scala.collection.mutable._


/**
  *
  *
  * @author DM
  *         2017
  * @version v1.0.0.1
  *
  *
  */
object VagueState {
  /**
    *
    */
  val STAR_CHAR = '*'
  val patternMap = new HashMap[String, List[Any]]
}

/**
  *
  * @tparam T
  */
trait VagueState[T] {
  /**
    *
    * @param t
    * @param nextNormalState
    */
  def addNextNormalStateMap(t: T, nextNormalState: VagueState[T]): Unit

  /**
    *
    * @param t
    * @param nextVagueState
    */
  def addNextVagueStateMap(t: T, nextVagueState: VagueState[T]): Unit

  /**
    *
    * @return
    */
  def getNextVagueStateMap: Map[T, VagueState[T]]

  def getNextNormalStateMap: Map[T, VagueState[T]]

  /**
    *
    * @param vagueResultEntityList
    */
  def addVagueResultEntityList(vagueResultEntityList: VagueResultEntity): Unit

  def getVagueResultEntityList: ListBuffer[VagueResultEntity]

  /**
    *
    * @param pattern
    * @param vagueResultEntity
    */
  def addState(pattern: String, vagueResultEntity: VagueResultEntity): Unit

  /**
    *
    * @param patterns
    * @param resultDataSet
    */
  def serachVagueResult(patterns: String, resultDataSet: Set[Any]): Unit
}
